pkg/ccl/backupccl/restore_online.go

```diff
@@ -140,6 +221,39 @@ func rewriteSpan(
 	return span, nil
 }
 
+// linkExternalFiles runs through all entries produced by genSpans and links in
+// all files in the entries rewritten using the passed rewriter. It assumes that
+// the target spans have already been split and scattered.
+func linkExternalFiles(
+	ctx context.Context,
+	execCtx sql.JobExecContext,
+	genSpans func(ctx context.Context, spanCh chan execinfrapb.RestoreSpanEntry) error,
+	kr KeyRewriter,
+	fromSystemTenant bool,
+	requestFinishedCh chan<- struct{},
+) (approxRows int64, approxDataSize int64, err error) {
+	ctx, sp := tracing.ChildSpan(ctx, "backupccl.linkExternalFiles")
+	defer sp.Finish()
+	defer close(requestFinishedCh)
+
+	log.Infof(ctx, "ingesting remote files")
+
+	workers := int(onlineRestoreLinkWorkers.Get(&execCtx.ExecCfg().Settings.SV))
+
+	grp := ctxgroup.WithContext(ctx)
+	ch := make(chan execinfrapb.RestoreSpanEntry, 4)
```

Comment:
I suppose. It'll still be blocked on processing workers just a little later / can take a tad longer to be rescheduled before we're blocked on it to get them work. Actually, maybe we should make this much bigger (100?) since genSpans is iterating the metadata ssts, so we'd kinda like to let it do as much of that as possible in one go?